src/lib.rs

```diff
@@ -32,15 +34,16 @@ pub trait Metric<K: ?Sized> {
 }
 
 /// A node within the [BK-tree](https://en.wikipedia.org/wiki/BK-tree).
-struct BKNode<K> {
+#[cfg_attr(feature = "serde-support", derive(serde_derive::Serialize, serde_derive::Deserialize))]
```

Comment:
What is the advantage of using `cfg_attr` over `derive` to add serialization support? Looks like we're both trying to make `BKTree` serializable with our PRs, [but I went with using `derive`](https://github.com/eugene-bulkin/rust-bk-tree/pull/17). I'm still learning Rust and am interested to know how your approach compares to mine. Thanks!

Reply:
My changes add a feature `serde-support` that includes the _optional_ serde dependencies. So the above also uses `serde_derive` but only if the `serde-support` feature has been enabled. Otherwise, folks that do not want/need serialization would still have to include those dependencies.

Pretty sure. It's been a while (and I consider myself to be new to rust also...still!!)

Reply:
Okay! That makes a lot of sense. You're PR is definitely more flexible for users, then. Thank you for taking time out of your day to explain that! Learned something new today :smile: